robot/rover/internal_comms/src/CommandCenter.cpp

```diff
@@ -4,167 +4,151 @@
-        #endif
-    }
+  /*Serial.print("Command id: ");
+  Serial.print(commandID, HEX);
+  Serial.print(" Argsize: ");
+  Serial.print(argumentSize, DEC);
+  Serial.print(" Bytes read after argsize ");
+  Serial.print(bytesRead, DEC);
+  Serial.print(" stopByte: ");
+  Serial.print(stopByte, HEX);
+  Serial.print("\n");*/
 
-    void CommandCenter::readCommand()
-    {
-        Command* command = CommandCenter::processCommand();
-//        Serial.write(command->rawArgsLength);
-//        for(int i = 0 ; i < command->rawArgsLength ;i++){
-//            Serial.write(command->rawArgs[i]);
-//        }
-
-        if(command->isValid)
-        {
-            this->executeCommand(command->commandID, command->rawArgs, command->rawArgsLength);
-        }
-        else
-        {
-            // Handle invalid command
-            // (Create error message and put it at the front of the queue so that the 
-            // TX2 can resend)
-        }
-
-        free(command->rawArgs);
-        //command->rawArgs = nullptr;
-
-        delete command;
-    }
+  if (bytesRead != argumentSize) {
+    // bad, but no way of dealing with it for now
+  }
 
-    void CommandCenter::sendDebug(const char* debugMessage)
-    {
-        Message* message = this->createMessage(COMMAND_DEBUG_MSG, strlen(debugMessage), (byte*)debugMessage);
-        this->queueMessage(*message);
-    }
+  if (stopByte != 0x0a) {
+    // bad, but no way of dealing with it for now
+  }
 
-    void CommandCenter::sendMessage() {
+  return cmd;
+}
 
-        if (digitalRead(enablePin)) {
-            if (!messageQueue.empty()) {
-                Message message = messageQueue.front();
-                messageQueue.pop();
+uint8_t CommandCenter::readArgSize() { return waitForSerial(); }
 
-                Serial.write(message.messageID);
-                Serial.write(message.rawArgsLength);
+Message* CommandCenter::createMessage(int messageID, int rawArgsLength,
+                                      byte* rawArgs) {
+  Message* message = (Message*)malloc(sizeof(Message));
+  message->messageID = messageID;
+  message->rawArgsLength = rawArgsLength;
+  message->rawArgs = malloc(rawArgsLength);
+  memcpy(message->rawArgs, rawArgs, rawArgsLength);
+  return message;
+}
 
-                if(message.rawArgsLength > 0)
-                    Serial.write(message.rawArgs, message.rawArgsLength);
+void CommandCenter::queueMessage(Message& message) {
+  // We assume that the queue will never fill up
+  // Thus, never fail...
+  messageQueue.push(message);
+}
 
-                Serial.write(0x0A);
+uint8_t CommandCenter::waitForSerial() {
+  unsigned long start = millis();
+  while (!Serial.available()) {
+    unsigned long current = millis() - start;
+    if (current > 50) break;
+  }
+  return Serial.read();
+}
 
+// No where else uses rxPin and txPin, BUT can't get rid of them without
+// breaking other code that is out of scope for me (Marc)
```

Comment:
It was designed like this for the PDS mcu but it might not be necessary anymore with PDS2. We will leave it since it is not causing any problems.